storage/storage.rs

```diff
@@ -203,13 +206,31 @@ impl<D> MVCCStorage<D> {
     where
         D: DurabilityService,
     {
-        for record in self.durability_service.iter_type_from_start::<CommitRecord>().unwrap() {
-            if let Ok((commit_sequence_number, commit_record)) = record {
-                self.write_commit_record(commit_sequence_number, commit_record)?;
+        if self.durability_service.previous() == SequenceNumber::MIN {
+            self.isolation_manager.load_aborted(0, SequenceNumber::MIN); // Initialise the watermark to zero
+        }
+
+        for record in IsolationManager::iterate_commit_status_from_disk(&self.durability_service, &SequenceNumber::MIN).unwrap() {
+            if let Ok(commit_status) = record {
+                match commit_status {
+                    // TODO: Relative indexes once sequence numbers become discontinuous
+                    CommitStatus::Applied(commit_sequence_number, commit_record) => {
+                        self.isolation_manager.load_applied(commit_sequence_number.number().number() as i64, commit_sequence_number, commit_record.into_owned());
+                    },
+                    CommitStatus::Closed(commit_sequence_number) => {
+                        self.isolation_manager.load_aborted(commit_sequence_number.number().number() as i64, commit_sequence_number);
+                    }
+                    CommitStatus::Pending(commit_sequence_number, commit_record) => {
+                        self.isolation_manager.opened(commit_record.open_sequence_number()); // try_commit currently decrements reader count.
+                        self.write_commit_record(commit_sequence_number, commit_record.into_owned())?;
+                    }
```

Comment:
TODO: Using the sequence number as relative_index fails as soon as relative_index =0 does not hold Sequence number 0.